Cython/Compiler/Pythran.py

```diff
@@ -0,0 +1,154 @@
+    if hasattr(Ty,"is_none") and Ty.is_none:
+        return "decltype(pythonic::__builtin__::None)"
+    if Ty.is_numeric:
+        return Ty.sign_and_name()
+    raise ValueError("unsupported pythran type %s (%s)" % (str(Ty), str(type(Ty))))
+    return None
+
+def type_remove_ref(ty):
+    return "typename std::remove_reference<%s>::type" % ty
+
+def pythran_binop_type(op, tA, tB):
+    return "decltype(std::declval<%s>() %s std::declval<%s>())" % \
+        (pythran_type(tA), op, pythran_type(tB))
+
+def pythran_unaryop_type(op, type_):
+    return "decltype(%sstd::declval<%s>())" % \
+        (op, pythran_type(type_))
+
+def pythran_indexing_type(type_, indices):
+    def index_code(idx):
+        if idx.is_slice:
+            if idx.step.is_none:
+                func = "contiguous_slice"
+                n = 2
+            else:
+                func = "slice"
+                n = 3
+            return "pythonic::types::%s(%s)" % (func,",".join(["0"]*n))
+        elif idx.type.is_int:
+            return "std::declval<long>()"
+        elif idx.type.is_pythran_expr:
+            return "std::declval<%s>()" % idx.type.pythran_type
+        raise ValueError("unsupported indice type %s!" % str(idx.type))
+    indexing = ",".join(index_code(idx) for idx in indices)
+    return type_remove_ref("decltype(std::declval<%s>()(%s))" % (pythran_type(type_), indexing))
+
+def pythran_indexing_code(indices):
+    def index_code(idx):
+        if idx.is_slice:
+            values = idx.start, idx.stop, idx.step
+            if idx.step.is_none:
+                func = "contiguous_slice"
+                values = values[:2]
+            else:
+                func = "slice"
+            return "pythonic::types::%s(%s)" % (func,",".join((v.pythran_result() for v in values)))
+        elif idx.type.is_int:
+            return idx.result()
+        elif idx.type.is_pythran_expr:
+            return idx.pythran_result()
+        raise ValueError("unsupported indice type %s!" % str(idx.type))
+    return ",".join(index_code(idx) for idx in indices)
+
+def pythran_func_type(func, args):
+    args = ",".join(("std::declval<%s>()" % pythran_type(a.type) for a in args))
+    return "decltype(pythonic::numpy::functor::%s{}(%s))" % (func, args)
+
+def to_pythran(op,ptype=None):
+    #if hasattr(op,"entry") and hasattr(op.entry, "pythran_buf") and op.entry.pythran_buf is not None:
+    #    return op.entry.pythran_buf.cname
+    op_type = op.type
+    if is_type(op_type,["is_pythran_expr","is_int","is_numeric","is_float","is_complex"]):
+        return op.result()
+    if op.is_none:
+        return "pythonic::__builtin__::None"
+    if ptype is None:
+        ptype = pythran_type(op_type)
+    assert(op.type.is_pyobject)
+    return "from_python<%s>(%s)" % (ptype, op.py_result())
+
+def from_pythran():
+    return "to_python"
+
+def is_type(type_, types):
+    for attr in types:
+        if hasattr(type_, attr) and getattr(type_, attr):
```

Comment:
Fixed in https://github.com/cython/cython/pull/1607/commits/e3fb29c91b13edcc9297b02b0ca10e9a3da3928d